src/algorithms/perlinnoise.jl

```diff
@@ -0,0 +1,116 @@
+"""
+    PerlinNoise
+    PerlinNoise(; kw...)
+
+Create a Perlin noise neutral landscape model with values ranging 0-1.
+
+# Keywords
+
+- `periods::Tuple{Int,Int}`: the number of periods of Perlin noise across row and 
+    column dimensions for the first octave.
+- `octaves::Int`: the number of octaves that will form the Perlin noise.
+- `lacunarity::Int` : the rate at which the frequency of periods increases for each octive.
+- `persistance::Real` : The rate at which the amplitude of periods decreases for each octive.
+- `valley::String`: The kind of valley bottom that will be mimicked: `:u` (the defualt) 
+    produces u-shaped valleys, `:v` produces v-shaped valleys, and `:-` 
+    produces flat bottomed valleys.
+"""
+Base.@kwdef struct PerlinNoise{PT<:Real} <: NeutralLandscapeMaker 
```

Comment:
maybe we should follow the same constructor convention throughout the repo

Reply:
Yeah - style in general probably.

Reply:
Agreed - but maybe @rafaqz approach is more user-friendly?

Reply:
We could revisit style once we have the bulk of things written and it's clear what the best way is. But I'm happy to change this to whatever is preffered